src/bacnet/basic/object/netport.c

```diff
@@ -2951,16 +2798,9 @@ static bool Network_Port_FD_BBMD_Address_Write(
                 *error_class = ERROR_CLASS_PROPERTY;
                 *error_code = ERROR_CODE_WRITE_ACCESS_DENIED;
                 break;
-            } else if (value->host.ip_address.length == 4) {
-                status = Network_Port_Remote_BBMD_IP_Address_Set(
-                    object_instance, value->host.ip_address.value[0],
-                    value->host.ip_address.value[1],
-                    value->host.ip_address.value[2],
-                    value->host.ip_address.value[3]);
-                if (status) {
-                    status = Network_Port_Remote_BBMD_BIP_Port_Set(
-                        object_instance, value->port);
-                }
+            } else if ( ((value->host_ip_address) && (value->host.ip_address.length == 4)) ||
+                      ((value->host_name) && (is_valid_hostname(&value->host.name))) ) {
+                status = Network_Port_Remote_BBMD_Address_Set(object_instance, value);
```

Comment:
Do we want to remove the if (status)... here?

Reply:
I am not sure what you mean exactly here. The below checks in or instance 2805 are required.

Reply:
It's coming up as outdated now, but before, there was in if statement being removed and I was just checking if you meant it, but seeing as it has since been removed, I don't think you did.